src/nile/starknet_cli.py

```diff
@@ -0,0 +1,115 @@
+"""Call the starknet_cli."""
+
+import io
+import sys
+from types import SimpleNamespace
+
+from starkware.starknet.cli import starknet_cli
+from starkware.starknet.cli.starknet_cli import NETWORKS
+
+from nile.common import ABIS_DIRECTORY, BUILD_DIRECTORY, GATEWAYS
+
+ARGS = [
+    "contracts",
+    "contract_address",
+    "inputs",
+    "signature",
+    "address",
+    "abi",
+    "sender",
+    "max_fee",
+    "mainnet_token",
+    "hash",
+]
+
+
+async def execute_call(cmd_name, network, **kwargs):
+    """Build and execute call to starknet_cli."""
+    args = set_args(network)
+    command_args = set_command_args(**kwargs)
+    cmd = getattr(starknet_cli, cmd_name)
+    return await capture_stdout(cmd(args=args, command_args=command_args))
+
+
+async def capture_stdout(func):
+    """Return the stdout during the passed function call."""
+    stdout = sys.stdout
+    sys.stdout = io.StringIO()
+    await func
+    output = sys.stdout.getvalue()
+    sys.stdout = stdout
+    result = output.rstrip()
+    return result
+
+
+def set_args(network):
```

Comment:
```suggestion
def set_context(network):
```
this seems more accurate according the description